File: harvester/models/tv_show.py
class TVShow:
    def __init__(self, tt_show_id, title, runtime):
        self.tt_show_id = tt_show_id
        self.title = title
        self.runtime = runtime
        self.episodes = []

    def add_episodes(self, episodes):
        self.episodes = episodes

    def to_dict(self):
        return {
            'tt_show_id': self.tt_show_id,
            'title': self.title,
            'runtime': self.runtime,
            'episodes': [episode.to_dict() for episode in self.episodes]
        }


class Episode:
    def __init__(self, tt_episode_id, title, number, aired, avg_rating, votes, season_id):
        self.tt_episode_id = tt_episode_id
        self.title = title
        self.number = number
        self.aired = aired
        self.avg_rating = avg_rating
        self.votes = votes
        self.season_id = season_id

    def to_dict(self):
        return {
            'tt_episode_id': self.tt_episode_id,
            'title': self.title,
            'number': self.number,
            'aired': self.aired,
            'avg_rating': self.avg_rating,
            'votes': self.votes,
            'season_id': self.season_id
        }

File: core/models/tvshow.go
package models

import (
	"strconv"
	"time"
)

type Popularity struct {
	ShowID      string `json:"show_id"`
	TimesViewed int    `json:"times_viewed"`
}

type TvShow struct {
	ShowID       string   `json:"show_id"`
	Title        string   `json:"title"`
	Runtime      int      `json:"runtime"`
	Votes        int      `json:"votes"`
	AvgRating    float64  `json:"avg_rating"`
	MedianRating float64  `json:"median_rating"`
	Seasons      []Season `json:"seasons"`
}

type Season struct {
	Number       int       `json:"number"`
	AvgRating    float64   `json:"avg_rating"`
	MedianRating float64   `json:"median_rating"`
	Votes        int       `json:"votes"`
	Episodes     []Episode `json:"episodes"`
}

type Episode struct {
	Number    int       `json:"number"`
	EpisodeID string    `json:"episode_id"`
	Title     string    `json:"title"`
	Aired     time.Time `json:"aired"`
	AvgRating float64   `json:"avg_rating"`
	Votes     int       `json:"votes"`
}

func (tvShow *TvShow) TvShowBuilder(tvShowDTO TvShowDTO) {
	tvShow.ShowID = tvShowDTO.ShowID
	tvShow.Title = tvShowDTO.Title
	tvShow.Runtime, _ = strconv.Atoi(tvShowDTO.Runtime)

	lastSeasonNumber := tvShowDTO.Episodes[len(tvShowDTO.Episodes)-1].SeasonID
	if lastSeasonNumber == -1 {
		lastSeasonNumber = tvShowDTO.Episodes[len(tvShowDTO.Episodes)-2].SeasonID
	}
	seasons := make([]Season, lastSeasonNumber)

	for currentSeason := 1; currentSeason <= lastSeasonNumber; currentSeason++ {
		for _, episode := range tvShowDTO.Episodes {
			if episode.SeasonID == currentSeason {
				seasons[currentSeason-1].Number = currentSeason
				seasons[currentSeason-1].Episodes = append(seasons[currentSeason-1].Episodes, Episode{
					Number:    episode.Number,
					EpisodeID: episode.EpisodeID,
					Title:     episode.Title,
					Aired:     episode.Aired.Time,
					AvgRating: episode.AvgRating,
					Votes:     episode.Votes,
				})
			}
		}
	}

	tvShow.Seasons = seasons
}
